src/main/java/net/dv8tion/jda/api/events/interaction/ModalSubmitInteractionEvent.java

```diff
@@ -0,0 +1,134 @@
+ */
+
+package net.dv8tion.jda.api.events.interaction;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.ItemComponent;
+import net.dv8tion.jda.api.interactions.components.text.TextInput;
+import net.dv8tion.jda.api.requests.restaction.interactions.ReplyCallbackAction;
+import net.dv8tion.jda.internal.interactions.ModalInteractionImpl;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.CheckReturnValue;
+import java.util.List;
+
+/**
+ * Indicates that a {@link net.dv8tion.jda.api.interactions.components.text.Modal Modal} was submitted.
+ *
+ * <h2>Requirements</h2>
+ * To receive these events, you must unset the <b>Interactions Endpoint URL</b> in your application dashboard.
+ * You can simply remove the URL for this endpoint in your settings at the <a href="https://discord.com/developers/applications" target="_blank">Discord Developers Portal</a>.
+ *
+ * @see net.dv8tion.jda.api.interactions.ModalInteraction
+ */
+public class ModalSubmitInteractionEvent extends GenericInteractionCreateEvent
+{
+    private final ModalInteractionImpl interaction;
+
+    public ModalSubmitInteractionEvent(@NotNull JDA api, long responseNumber, @NotNull ModalInteractionImpl interaction)
+    {
+        super(api, responseNumber, interaction);
+        this.interaction = interaction;
+    }
+
+    @NotNull
+    @Override
+    public ModalInteractionImpl getInteraction()
+    {
+        return interaction;
+    }
+
+    /**
+     * Returns the custom id of the Modal in question
+     * @return Custom id
+     */
+    @NotNull
+    public String getModalId()
+    {
+        return interaction.getModalId();
+    }
+
+    /**
+     * Returns a List of {@link ActionRow ActionRows} the modal in question contains
+     *
+     * Contains information like the text the user entered on {@link TextInput TextInputs}
+     *
+     * @return List of {@link ActionRow ActionRows}
+     */
+    @NotNull
+    public List<ActionRow> getComponents()
+    {
+        return interaction.getComponents();
+    }
+
+    /**
+     * Convenience method to get a {@link TextInput TextInput} by its id from the List of components.
+     *
+     * Returns null if no TextInput with that id has been found
+     *
+     * @param id The custom id
+     *
+     * @return TextInput with this id, or null
+     */
+    @Nullable
+    public TextInput getInputField(String id)
+    {
+        for (ActionRow row : getComponents())
+        {
+            for (ItemComponent component : row)
+            {
+                if (component instanceof TextInput)
+                {
+                    TextInput textInput = (TextInput) component;
+                    if (textInput.getId().equals(id))
+                        return textInput;
+                }
+            }
+        }
```

Comment:
```suggestion
        return getComponents().stream()
                .map(ActionRow::getComponents)
                .filter(TextInput.class::isInstance)
                .map(TextInput.class::cast)
                .filter(textInput -> textInput.getId().equals(id))
                .findFirst().orElse(null);
```

Just a suggestion, whenever you prefer this is fully up-to you